### ISSUE TITLE
Publish dual-format package

### ISSUE THREAD

**Issue by karashiiro** [the-convocation/twitter-scraper#22]
As a better solution for #4, this package should be published in both CJS and ESM formats for convenient usage in both backend and frontend environments. The package could be changed to force ESM, but I believe that would be a callous solution for backend developers. Node.js supports ESM with dynamic imports and/or the `"type": "module"` specifier, but not all frameworks do, or at least not well (Nest.js comes to mind, and TypeScript's handling of import paths is unacceptable for established projects).

Beware the [dual package hazard](https://nodejs.org/api/packages.html#dual-commonjses-module-packages), of course.

**Comment by Owen3H** [comment]
I suggest maybe using Rollup + ESBuild to generate both UMD and ESM bundles.
Heres an example I use in one of my existing projects.

`package.json`
```json
"main": "dist/index.js",
"module": "dist/index.mjs",
"types": "dist/types.d.ts",
"scripts": {
  "clean": "rimraf dist",
  "rollup": "npm run clean && rollup -c",
  "build": "npm run rollup && npm run typedoc",
  "typedoc": "npx typedoc --options typedoc.json"
},
"devDependencies": {
  "@rollup/plugin-json": "^6.0.0",
  "@types/node": "^20.3.1",
  "rimraf": "^5.0.1",
  "rollup-plugin-dts": "^5.3.0",
  "rollup-plugin-esbuild": "^5.0.0",
  "typedoc": "^0.24.8",
  "typedoc-theme-hierarchy": "^4.0.0",
  "typescript": "^5.1.6"
}
```

`rollup.config.mjs`
```js
import pkg from './package.json' assert { type: 'json' }

import esbuild from 'rollup-plugin-esbuild'
import dts from 'rollup-plugin-dts'
import json from '@rollup/plugin-json'

const generatedCode = {
    arrowFunctions: true,
    constBindings: true,
    objectShorthand: true
}

const source = {
    input: 'src/index.ts',
    external: ['undici', 'tslib', 'net'],
    plugins: [esbuild(), json()],
    output: [{ 
        generatedCode, 
        file: pkg.main, 
        format: 'cjs' 
    }, {
        generatedCode,
        file: pkg.module, 
        format: 'es' 
    }]
}

const types = {
    input: 'src/index.ts',
    output: [{ file: 'dist/types.d.ts', format: 'es' }],
    plugins: [dts()]
}

export default [source, types]
```

**Comment by karashiiro** [comment]
That's normally what I would do, but that can create issues for downstream bundlers and test frameworks since that's just creating two completely separate builds which can be loaded independently (see the documentation link in my other comment).

The `module` field is also nonstandard, and I think browser bundlers don't always respect it; `exports` is a more reliable way to configure this.

**Comment by Owen3H** [comment]
Yeah I see your point, I am currently writing [undici-shim](https://github.com/Owen3H/undici-shim) which tries to achieve the same thing using the following approach.

```json
"import": {
  "node": "./dist/esm/node.js",
  "default": "./dist/esm/browser.js"
},
"require": {
  "node": "./dist/umd/node.cjs",
  "default": "./dist/umd/browser.cjs"
}
```

This way, you would need to tell rollup to use the `.cjs` extension for the UMD build.

**Comment by Owen3H** [comment]
Yeah I see your point, I am currently writing [undici-shim](https://github.com/Owen3H/undici-shim) which tries to achieve the same thing using the following approach.
Although this way, you would need to tell rollup to use the `.cjs` extension for the UMD build.


```json
"import": "./dist/esm/index.js",
"require": "./dist/umd/index.cjs"

```

**Comment by karashiiro** [comment]
Closed with #96 as this is required for #89, can improve later
